### PR TITLE
Load grub modules on mem init

### PR DESCRIPTION
**Pull request by SamTebbs33**
This patch stores the modules loaded by grub at boot time. Depends on #105

### CODE REVIEW

File: src/kernel/arch/x86/arch.zig [hunk 1]
```diff
@@ -215,7 +216,7 @@ pub fn haltNoInterrupts() noreturn {
 ///     IN comptime options: type         - The build options that is passed to the kernel to be
 ///                                         used for run time testing.
 ///
-pub fn init(mem_profile: *const MemProfile, allocator: *Allocator, comptime options: type) void {
+pub fn init(mb_info: *multiboot.multiboot_info_t, mem_profile: *const MemProfile, allocator: *Allocator) void {
```
[the remark this answers is not in the document]
Comment (SamTebbs33) [reply src/kernel/arch/x86/arch.zig]: I don't know what you're talking about...

File: src/kernel/mem.zig [hunk 2]
```diff
@@ -15,17 +18,83 @@ extern var KERNEL_VADDR_START: *u32;
...
+    ADDR_OFFSET = @ptrToInt(&KERNEL_ADDR_OFFSET);
+    const mem_profile = MemProfile{
         .vaddr_end = @ptrCast([*]u8, &KERNEL_VADDR_END),
         .vaddr_start = @ptrCast([*]u8, &KERNEL_VADDR_START),
         .physaddr_end = @ptrCast([*]u8, &KERNEL_PHYSADDR_END),
         .physaddr_start = @ptrCast([*]u8, &KERNEL_PHYSADDR_START),
         // Total memory available including the initial 1MiB that grub doesn't include
         .mem_kb = mb_info.mem_upper + mb_info.mem_lower + 1024,
         .fixed_alloc_size = FIXED_ALLOC_SIZE,
+        .boot_modules = @intToPtr([*]multiboot.multiboot_mod_list, physToVirt(mb_info.mods_addr))[0..mods_count],
```
[the remark this answers is not in the document]
Comment (SamTebbs33) [reply src/kernel/mem.zig]: Nice idea

[the remark this answers is not in the document]
Comment (DrDeano) [reply src/kernel/mem.zig]: If you can have zero elements, I think it is fine. I didn't know if [0..0] would work.

[the remark this answers is not in the document]
Comment (SamTebbs33) [reply src/kernel/mem.zig]: It does seem to work :)

**Pull request merged**